packages/databyss-ui/components/Editor/slate/markup.js

```diff
@@ -0,0 +1,78 @@
+import cloneDeep from 'clone-deep'
+import { Editor, Value } from 'slate'
+
+export const stateToSlate = (slate, _id) => {
+  const { nodes } = slate
+  let text = ''
+  let ranges = nodes.map((n, i) => {
```

Comment:
Let's extract this to a fn called `getRangesFromBlock` so that we can use it here and in `ContentEditable`